src/pilot/config_idrac.py

```diff
@@ -204,6 +207,199 @@ def configure_bios_nics_boot_settings(drac_client, ip_service_tag, pxe_nic_id):
+    """Set the legacy, non-UEFI, boot protocol of a NIC to PXE.
+
+    If successful, the pending value of the NIC's legacy boot
+    protocol attribute is set. For the new value to be applied, a
+    configuration job must be created and the node must be rebooted.
+
+    :param nic_id: id of the network interface controller (NIC)
+    :param drac_client: drac_client from python-dracclient
+    :returns: dictionary containing a 'commit_required' key with a
+              boolean value indicating whether a configuration job
+              must be created for the new legacy boot protocol
+              setting to be applied
+    :raises: WSManRequestFailure on request failures
+    :raises: WSManInvalidResponse when receiving invalid response
+    :raises: DRACOperationFailed on error reported back by the iDRAC
+             interface
+    """
+    return set_nic_legacy_boot_protocol(nic_id, 'PXE', drac_client)
+
+
+def _list_nic_settings(selection_expression):
+    result = {}
+    configurable_attributes = [
+        (uris.DCIM_NICEnumeration,
+         'DCIM_NICEnumeration',
+         nic.NICEnumerationAttribute),
+        (uris.DCIM_NICString,
+         'DCIM_NICString',
+         nic.NICStringAttribute),
+        (uris.DCIM_NICInteger,
+         'DCIM_NICInteger',
+         nic.NICIntegerAttribute)]
+
+    for (resource, class_name, attr_cls) in configurable_attributes:
+        attribs = _get_nic_config(resource, class_name,
+                                  selection_expression, attr_cls)
+
+        if not set(result).isdisjoint(set(attribs)):
+            raise exceptions.DRACOperationFailed(
+                drac_messages=('Colliding attributes %r' % (
+                    set(result) & set(attribs))))
+
+        result.update(attribs)
+
+    return result
+
+
+def get_nic_setting(nic_id, attribute_name):
+    """Obtain a setting of a NIC.
+
+    :param nic_id: id of the network interface controller (NIC)
+    :param attribute_name: name of the setting
+    :returns: value of the attribute on successful query, None
+              otherwise
+    :raises: WSManRequestFailure on request failures
+    :raises: WSManInvalidResponse when receiving invalid response
+    :raises: DRACOperationFailed on error reported back by the iDRAC
+             interface
+    :raises: NotFound when no settings for NIC found
+    """
+    selection_expression = ('InstanceID = '
+                            '"%(fqdd)s:%(attribute_name)s"') % {
+        'fqdd': nic_id,
+        'attribute_name': attribute_name}
+    settings = _list_nic_settings(selection_expression)
+
+    # Were no settings found?
+    if not settings:
+        raise exceptions.NotFound(
+            what=('settings for NIC %(nic)s') % {
+                'nic': nic_id})
+
+    # Do the settings include the attribute?
+    if attribute_name not in settings:
+        return None
+
+    return settings[attribute_name]
+
+
+def set_nic_setting(nic_id, attribute_name, value, drac_client):
```

Comment:
We should be using the upstreamed set_nic_settings() method and delete this one.